File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n_row=matrix.length;
        int n_col=matrix[0].length;
        for(int i=0;i<n_row;i++){
            if(matrix[i][0]<=target && matrix[i][n_col-1]>=target){
                for(int j:matrix[i]){
                    if(j==target)return true;
                }
            }
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row_sz = matrix.size();
        int col_sz = matrix[0].size();
        int row = 0;
        int col = col_sz - 1;
        while (col >= 0 && row < row_sz) {
            if (matrix[row][col] == target) {
                return true;
            } else if (matrix[row][col] > target) {
                col--;
            } else if (matrix[row][col] < target) {
                row++;
            }
        }
        return false;
    }
};

